keps/993-two-phase-admission/README.md

```diff
@@ -253,6 +254,27 @@ The controller implementing a particular check should:
 * After approving the workload, keep an eye on the check if it starts failing,
   fail the check and cause the workload to move back to the suspended state.
 
+### Preemption Admission Check Controller
+
+Because now, the time when the eviction of the preemption candidate should vary based on the preemptor state
+the scheduler will not issue the eviction during it's process instead it will set a `Pending` admission check
+that is manged by a new built-in admission check controller.
+
+The **Admission Check Controller** will:
+
+- Watch for a change in state of the workloads pending preemption.
+- Watch for workloads that are finishing execution and therefore releasing quota.
+- Watch for AdmissionCheck changes, since the preemption policy can change.
+
+Since evaluating the current preemption state and the needs of a workload requires checking the sate of all resource pool the events are rate limited the kueue's cache is used.
+
+At every run the controller will get the list of workloads pending preemption.
+Since for some of these workloads is not necessary to issue eviction at that given point (eg. Having a pending check that uses AfterCheckPassedOrOnDemand policy) their quota reservation will be ignored.
+Fore every other preemption pending workloads, it will check if it can fit without evicting other workloads, case in which the preemption admission check condition will be set to `Ready`.
+If eviction of other workload is still needed, an updated list candidates is created and eviction is issued for all of them.
+If the updated list of candidates is empty, meaning that the preemption can no longer succeed, the preemption admission check is set as `Retry`, the workload will louse it's quota reservation and requeued.
```

Comment:
```suggestion
If the updated list of candidates is empty, meaning that the preemption can no longer succeed, the preemption admission check is set as `Retry`, the workload will lose it's quota reservation and requeued.
```

This might be a good reason for have preemption as an AdmissionCheck, if having a permanent record for when preemptions were issued, or were not possible.

But is such permanent record useful? Given the continuous nature of preemptions, events might be a better fit. We would like to say:
1. preemptions were issued
2. preemptions were not possible.

(1) is completely continuous, so not a good fit for conditions or admission check, because lastTransitionTime wouldn't change if we issue more and more preemptions.
(2) is more permanent: once issued, we would like to requeue. But this could be represented by setting the condition `QuotaReserved` to false with reason `UnsatisfiablePreemptions`.

Reply:
I believe that events are a better for this.

Reply:
(misspell done)

Reply:
sounds good, then I'll consider that a "record" alone is not a sufficient to decide to implement preemptions as an AdmissionCheck.